src/ai/ChatHistorySource.js

```diff
@@ -0,0 +1,44 @@
+import { ContextSource } from './ContextSource.js';
+
+/**
+ * @class ChatHistorySource
+ * This stores the history of messages (user messages, assistant responses, and system function call messages)
+ * so that they can be provided as context to the ChatInterface.
+ * @todo: still needs some way to limit the max number of tokens sent as context
+ */
+export class ChatHistorySource extends ContextSource {
+    constructor() {
+        super('ChatHistory');
+
+        this.history = [];
+
+        // TODO: implement something like this.
+        //  But not too short, or a complex sequence of function calls might get cropped out too soon.
+        // this.PAST_MESSAGES_INCLUDED = 20;
+    }
+
+    addToHistory(role, content, functionCall) {
+        console.log(`added message { role: ${role}, content: ${content} } to chat history`);
```

Comment:
Should remove this log statement before merging in.

Reply:
Oof yes I have a lot to clean up 😅  I will make sure to give the entire thing another pass before merging